### PR TITLE
MINOR: Update docs for KIP-415

### PR DESCRIPTION
**Pull request by kkonstantine**
* Details are added under Connect's section to describe the behavior under the new Connect protocol
* Incremental cooperative rebalancing is added in the notable changes for 2.3.0

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by kkonstantine** [comment]
@rhauch @cmccabe @ewencp @hachikuji 
These are exclusively site-doc updates for KIP-415 and the upcoming release notes. 
I'd appreciate a review.

File: docs/connect.html [hunk 1]
```diff
@@ -508,7 +508,24 @@ <h4><a id="connect_administration" href="#connect_administration">Kafka Connect
...
+    When a connector is first submitted to the cluster, a rebalance is triggered between the Connect workers in order to distribute the load that consists of the tasks of the new connector.
+    This same rebalancing procedure is also used when connectors increase or decrease the number of tasks they require, when a connector's configuration is changed, or when a
+    worker is added or removed from the group as part of an intentional upgrade of the Connect cluster or due to a failure.
+    In versions prior to 2.3.0, the Connect workers would rebalance the full set of connectors and their tasks in the cluster as a simple way to make sure that each worker has approximately the same amount of work.
+    This behavior can be still enabled by setting <code>connect.protocol=eager</code>.
+    Starting with 2.3.0, Kafka Connect is using by default a protocol that performs
```
Comment (rhauch) [inline docs/connect.html:522]: Should this start a new paragraph? In fact, it seems like there might be a few paragraphs here.

File: docs/connect.html [hunk 2]
```diff
@@ -508,7 +508,24 @@ <h4><a id="connect_administration" href="#connect_administration">Kafka Connect
...
+    In case a worker leaves the group, intentionally or due to a failure, rebalancing of its tasks is triggered only after a delay equal to the value of the property <code>scheduled.rebalance.max.delay.ms</code>.
+    This delay, that has a default value of five minutes (<code>300000ms</code>), aims to provide a way to tolerate failures or upgrades of workers without immediately redistributing the load of a departing worker.
+    If this worker returns within the configured delay it gets its previous assigned tasks in full. For as long as <code>scheduled.rebalance.max.delay.ms</code> is in effect, these tasks remain unassigned.
+    If a worker does not return in time, the tasks get distributed among the remaining workers in the Connect cluster.
+
+    The new Connect protocol is enabled when all the workers that form the Connect cluster are configured with <code>connect.protocol=compatible</code>. Upgrade to the new Connect protocol is automatic
+    by default workers upgrade to 2.3.0. A rolling upgrade of the Connect cluster will activate incremental cooperative rebalancing when the last worker joins on version 2.3.0.
```
Comment (rhauch) [inline docs/connect.html]: This section seems a bit confusing. First, it mentions setting `connect.protocol=compatible`, but then it says it's automatic. Might want to clarify this a bit further to describe what the default is and what needs to change if they want to use the new rebalance logic.

File: docs/connect.html [hunk 3]
```diff
@@ -508,7 +508,24 @@ <h4><a id="connect_administration" href="#connect_administration">Kafka Connect
...
+    when connectors and tasks need to be distributed among the Connect workers. Balanced load among workers is still preserved under incremental cooperative rebalancing.
+    But in this case, a task is stopped temporarily during a rebalance only when it needs to move between workers. This transition might be immediate after consecutive lightweight rebalances between workers when a new worker joins.
+    In case a worker leaves the group, intentionally or due to a failure, rebalancing of its tasks is triggered only after a delay equal to the value of the property <code>scheduled.rebalance.max.delay.ms</code>.
+    This delay, that has a default value of five minutes (<code>300000ms</code>), aims to provide a way to tolerate failures or upgrades of workers without immediately redistributing the load of a departing worker.
+    If this worker returns within the configured delay it gets its previous assigned tasks in full. For as long as <code>scheduled.rebalance.max.delay.ms</code> is in effect, these tasks remain unassigned.
+    If a worker does not return in time, the tasks get distributed among the remaining workers in the Connect cluster.
```
Comment (rhauch) [inline docs/connect.html]: Some suggestions:

> ... <a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-415%3A+Incremental+Cooperative+Rebalancing+in+Kafka+Connect">incremental cooperative rebalancing</a>
that incrementally balances the connectors and tasks across the Connect workers, affecting only tasks that are new, to be removed, or need to move from one worker to another. Other tasks are not stopped and restarted during the rebalance, as they would have been with the old protocol.
>
> If a Connect worker leaves the group, intentionally or due to a failure, Connect waits for <code>scheduled.rebalance.max.delay.ms</code> before triggering a rebalance. This delay defaults to five minutes (<code>300000ms</code>) to tolerate failures or upgrades of workers without immediately redistributing the load of a departing worker. If this worker returns within the configured delay, it gets its previous assigned tasks in full. However, it does mean that the tasks may remain unassigned until the time specified by <code>scheduled.rebalance.max.delay.ms</code> elapses. If a worker does not return within that time limit, Connect will reassign those tasks among the remaining workers in the Connect cluster.

File: docs/connect.html [hunk 4]
```diff
@@ -508,7 +508,24 @@ <h4><a id="connect_administration" href="#connect_administration">Kafka Connect
...
+
+    The new Connect protocol is enabled when all the workers that form the Connect cluster are configured with <code>connect.protocol=compatible</code>. Upgrade to the new Connect protocol is automatic
+    by default workers upgrade to 2.3.0. A rolling upgrade of the Connect cluster will activate incremental cooperative rebalancing when the last worker joins on version 2.3.0.
+
+    You can use the REST API to view the current status of a connector and its tasks, including the id of the worker to which each was assigned. For example, querying the status of a file source (using <code>GET /connectors/file-source/status</code>) might produce output like the following:
```
Comment (rhauch) [inline docs/connect.html]: Maybe:
> The REST API exposes the current status of a connector and its tasks, including the ID of the worker to which each was assigned. For example, the <code>GET /connectors/file-source/status</code> shows the status of a connector named `file-source`:

**Review by rhauch — COMMENTED** [review]
Thanks, @kkonstantine. Very nice! I've left a few suggestions below:

File: docs/connect.html (hunk 1 above)
Comment (kkonstantine) [reply docs/connect.html:522]: certainly

File: docs/connect.html (hunk 3 above)
Comment (kkonstantine) [reply docs/connect.html]: Added almost unchanged. Let's re-read.

File: docs/connect.html (hunk 2 above)
Comment (kkonstantine) [reply docs/connect.html]: Indeed. Tried to improve.

File: docs/connect.html (hunk 4 above)
Comment (kkonstantine) [reply docs/connect.html]: Reused mostly the 2nd sentence of your suggestion. I'd like to retain `You can` (this btw was an old sentence I left unchanged)

**Comment by kkonstantine** [comment]
Thanks @rhauch !
I updated the PR after incorporating your suggestions and taking a few more passes. 
Anything missing or additional improvements?

**Review by rhauch — DISMISSED** [review]
LGTM. Thanks, @kkonstantine!

**Comment by rhauch** [comment]
@kkonstantine this PR should be against `trunk` rather than `2.3`. Then I'll backport to the `2.3` branch, and it will make the next 2.3.x release (maybe 2.3.1).

**Comment by kkonstantine** [comment]
That's right. I was focusing on the version related changes. 
Updated and rebased now. Backporting should exclude the 2.4.0 notes. Thanks @rhauch !

**Review by rhauch — APPROVED** [review]
LGTM. Thanks, @kkonstantine

**Pull request merged**